src/pieces/api/assets.py

```diff
@@ -156,56 +155,48 @@ def delete_asset_by_id(asset_id):
+        print(f"reclassify the asset to {classification} successfully")
+    except Exception as e:
+        show_error("Error reclassifying asset: ",{e})
 
-    working_asset = get_asset_by_id(asset_id)
-    # file_name = working_asset.get('name')
-    file_name = "Pieces_is_the_Best.tex" ################ TEST
-
-    # Define the path to the file
-    file_path = f"opened_snippets/{file_name}"
 
-    # Open and read the file content
+def update_asset(file_path,asset_id):
     try:
-        with open(file_path, 'r') as file:
-            new_content = file.read()
+        with open(file_path,"r") as f:
+            data = f.read()
     except FileNotFoundError:
-        print(f"File not found: {file_path}")
+        show_error("Error in update asset","File not found")
         return
-    except Exception as e:
-        show_error("Error reading file: ",{e})
+    asset_api = pos_client.AssetApi(api_client)
+    format_api = pos_client.FormatApi(api_client)
+
+    # get asset
+    created = asset_api.asset_snapshot(asset_id, transferables=False)
+
+    # update the original format's value
+    original = format_api.format_snapshot(created.original.id, transferable=True)
+    original_value = original.fragment.string.raw if original.fragment and original.fragment.string else None
+
+    # check if the string value is not empty
+    if original_value is None:
+        show_error("Error in update asset","Original value is empty")
         return
 
-    def update_string(obj):
-        if isinstance(obj, dict):
-            for key, value in obj.items():
-                if isinstance(value, dict):
-                    update_string(value)
-                elif isinstance(value, list):
-                    for item in value:
-                        update_string(item)
-                elif key == 'raw' and isinstance(value, str):
-                    obj[key] = new_content
-
-    # print(working_asset)
-    # exported_asset = ExportedAsset(name="This is a test", description="Testing description", raw=FileFormat(string={"string": "This, This is me testing export asset"}), created=GroupedTimestamp(value=datetime.now()))
-
-    # print(exported_asset)
-    format_api_instance = pos_client.FormatApi(api_client)
-    format = (
-        pos_client.Format(
-            asset=working_asset,
-            id=asset_id,
-            creator="ea47fe2f-a503-4edb-861a-7c55ca446859",
-            classification=Classification(generic="CODE", specific="tex"),
-            role="BOTH",
-            application=application,
-        ),
-    )
-    api_response = format_api_instance.format_update_value(
-        transferable=False, format=format
-    )
-    print(api_response)
\ No newline at end of file
+    # call our endpoint to update our value. && update our value.
+    original.fragment.string.raw = data
+    original = format_api.format_update_value(transferable=True, format=original)
```

Comment:
we dont necessarily need transferable:true here, because we are not using this value after the fact, meaning that we are not cacheing this value, and we are not displaying it to the user so the transferable can be false